File: withnode/4-chat-with-memory.js
import { ChatOllama } from "@langchain/ollama";
import { ConversationChain } from "langchain/chains";
import { BufferMemory } from "langchain/memory";

const model = new ChatOllama({model: "llama2"});
const memory = new BufferMemory({
    returnMessage: true
})
const conversation = new ConversationChain({
    llm: model,
    memory: memory,
    verbose: false
});


var result = await conversation.call({ input: "Hi, my name is Caleb. Remember that my name is Caleb."});
console.log(result);

result = await conversation.call({ input: "What is 1+1?"});
console.log(result);

result = await conversation.call({ input: "What is my name?"});
console.log(result);

File: withpython/4-chat-with-memory.py
from langchain.chains import ConversationChain
from langchain.memory import ConversationBufferMemory
from langchain_ollama import ChatOllama

llm = ChatOllama(model="llama2")
memory = ConversationBufferMemory()
conversation = ConversationChain(
    llm=llm,
    memory=memory,
    verbose=True
)

result = conversation.predict(input="Hi, my name is Caleb")
print(result)

result = conversation.predict(input="What is 1+1?")
print(result)

result = conversation.predict(input="What is my name?")
print(result)
